src/main/java/mServer/crawler/sender/arte/MediathekArte_de.java

```diff
@@ -64,210 +63,193 @@
-            
-            // erste Seite laden
-            ArteCategoryFilmsDTO dto = loadSubCategoryPage(gson, aUrl);
-            if(dto != null) {
-                // weitere Seiten laden und zu programId-liste des ersten DTO hinzufügen
-                for(int i = 1; i < dto.getPages(); i++) {
-                    String url = String.format(URL_SUBCATEGORY, LANG_CODE.toLowerCase(), aCategory, i);
-                    ArteCategoryFilmsDTO nextDto = loadSubCategoryPage(gson, url);
-                    if(nextDto != null) {
-                        nextDto.getProgramIds().forEach(programId -> dto.addProgramId(programId));
-                    }
-                }
-
-                // alle programIds verarbeiten
-                ListeFilme loadedFilme = loadPrograms(dto);
-                loadedFilme.forEach((film) -> {
-                    addFilm(film);
-                });
-            }
+      // erste Seite laden
+      int i = 2;
+      ArteCategoryFilmsDTO dto = loadSubCategoryPage(gson, aUrl);
+      if (dto != null) {
+        ArteCategoryFilmsDTO nextDto = dto;
+        while (nextDto != null && nextDto.hasNextPage()) {
+
+          // weitere Seiten laden und zu programId-liste des ersten DTO hinzufügen
+          String url = String.format(URL_SUBCATEGORY, LANG_CODE.toLowerCase(), aCategory, i);
+          nextDto = loadSubCategoryPage(gson, url);
+          if (nextDto != null) {
+            nextDto.getProgramIds().forEach(programId -> dto.addProgramId(programId));
+          }
+
+          i++;
         }
 
-        private ListeFilme loadPrograms(ArteCategoryFilmsDTO dto) {
-            ListeFilme listeFilme = new ListeFilme();
-
-            Collection<DatenFilm> futureFilme = new ArrayList<>();
-            dto.getProgramIds().forEach(programId -> {
-              try {
-                futureFilme.add(new ArteProgramIdToDatenFilmCallable(programId, LANG_CODE, getSendername()).call());
-              } catch (Exception exception) {
-                LOG.error("Es ist ein Fehler beim lesen der Arte Filme aufgetreten.", exception);
-              }
-            });
-            
-            CopyOnWriteArrayList<DatenFilm> finishedFilme = new CopyOnWriteArrayList<>();
-            futureFilme.parallelStream().forEach(finishedFilm -> {
-                try {
-                    if(finishedFilm != null)
-                    {
-                        finishedFilme.add(finishedFilm);
-                    }
-                } catch(Exception exception)
-                {
-                    LOG.error("Es ist ein Fehler beim lesen der Arte Filme aufgetreten.", exception);
-                }
-
-                });
-
-            listeFilme.addAll(finishedFilme);
-            return listeFilme;
+        // alle programIds verarbeiten
+        ListeFilme loadedFilme = loadPrograms(dto);
+        loadedFilme.forEach((film) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "film" parameter (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)